src/output-json.c

```diff
@@ -152,6 +152,86 @@ json_t *JsonAddStringN(const char *string, size_t size)
     return SCJsonString(tmpbuf);
 }
 
+void EveFileInfo(JsonBuilder *jb, const File *ff, const bool stored)
+{
+    size_t filename_size = ff->name_len * 2 + 1;
+    char filename_string[filename_size];
+    BytesToStringBuffer(ff->name, ff->name_len, filename_string, filename_size);
+    jb_set_string(jb, "filename", filename_string);
+
+    jb_open_array(jb, "sid");
+    for (uint32_t i = 0; ff->sid != NULL && i < ff->sid_cnt; i++) {
+        jb_append_uint(jb, ff->sid[i]);
+    }
+    jb_close(jb);
+
+#ifdef HAVE_MAGIC
+    if (ff->magic)
+        jb_set_string(jb, "magic", (char *)ff->magic);
+#endif
+    jb_set_bool(jb, "gaps", ff->flags & FILE_HAS_GAPS);
+    switch (ff->state) {
+        case FILE_STATE_CLOSED:
+            JB_SET_STRING(jb, "state", "CLOSED");
+#ifdef HAVE_NSS
+            if (ff->flags & FILE_MD5) {
+                size_t x;
+                int i;
+                char str[256];
+                for (i = 0, x = 0; x < sizeof(ff->md5); x++) {
+                    i += snprintf(&str[i], 255-i, "%02x", ff->md5[x]);
```

Comment:
@jasonish think it would make sense to have a `jb_set_hexstring_from_bytes` here? Think this loop of `snprintf` calls is relatively expensive.

Reply:
Alternatively, we can recode things to eliminate the `snprintf` loop.